\section{Introduction} \label{sec:gen-intro}

Humans have been breeding plants for food since the dawn of agriculture. This process
emerged from a haphazard form of artifical selection where farmers would save seed from the 
highest yielding, healthiest, and easiest to harvest plants to harvest the following year. 
Over thousands of years, this process produced many of the subspecies of plants of  
agricultural value today.

% Wikipedia Green revolution article.
Between 1930 and 1960, academic agricultural and plant breeding communities were setting the
stage for agricultural giants such as Norman Borlaug, Henry A. Wallace, and others explored 
options to breed plants for specific traits, and apply rigorous, scientifically-motivated 
management practices which improve varieties and the yield of agruculture practice worldwide.

As an example, in 1930, corn was an open-pollinated crop with an average 
yield around 20-30 bu/ac. By 1960, field corn had was usually single-cross hybrids 
producing twice the yield of those bred only two or three decades earlier. The implementation
agricultural practices that would drive the green revolution had been kicked off, and
the stage was set for a rapid improvement in yield and genetic gain year-over-year in
corn as well as many other crops. This improvement continued into the 2000s \citep{evenson2003}.

The green revolution drove crop production yields higher through a combination of 
improved agricultural practices and varieties that had greater yield potential
and agronomic stability. By the year 2000, advances in agronomy were slowing, and 
molecular breeding technologies were promising the next advancement in crop yield
through a focus on a better understanding of crop DNA. The advent of Bt Corn was the
landmark technology delivering on the promise of biotechnology in agriculture.
Bt corn was first commercialized in in 1996 and accounted for 63\% of U.S. grain 
corn by 2010 \citep{fernandez2012}. 

Biotechnology in agriculture is not limited only to transgenic traits. The cost of 
DNA sequencing and genetic marker assays began to fall during as the green revolution
came to a close. The cost of SNP assays has dropped so sharply that today it is 
possible to apply them to many thousands of samples in large breeding programs \citep{hiremath2012}. 
In many cases, it is less costly to genotype a seed than to grow it and observe its phenotype.

Because a SNP assay captures the genes and thus a part of the genetic potential of an 
individual, it is be possible to use the outcome of a SNP assay to predict the genetic
potential of the sampled individual in a affordable way. This process is known 
as genomic prediction, and when used to inform breeding 
program selections is known as genomic selection. Genomic prediction is typically 
applied early in a breeding program as a way to increase the overall selection 
pressure in a breeding program to increase genetic gain. A review of literature
and methods to perform this technique is presented in Section \ref{sec:lit-review}. 

The genetic gain in a breeding program is directly related to a host of factors. A 
general guideline to expected genetic gain each year in an inbred breeding 
program is presented in Equation \ref{eq:genetic_gain} \citep{fehr1987}. The equation is
often used as a general guideline to explore whether changes in breeding strategy will
have a positive or negative effect on genetic gain. 

\begin{equation} \label{eq:genetic_gain}
\begin{split}
    G_y           &= \textrm{genetic gain per year} \\
    k             &= \textrm{selection differential} \\
    r             &= \textrm{number of replications} \\
    t             &= \textrm{number of environments} \\
    n             &= \textrm{plants per plot} \\
    \sigma^2_{A}  &= \textrm{additive genetic variation} \\
    \sigma^2_{u}  &= \textrm{within-plot environmental variation} \\
    \sigma^2_{wg} &= \textrm{within-plot genetic variation} \\
    \sigma^2      &= \textrm{between-plot variation} \\
    \sigma^2_{ge} &= \textrm{genotype by environment interaction variation} \\
    \sigma^2_{g}  &= \textrm{genotypic variation} \\
    G_y           &= \frac{k\sigma^2_A}{y \sqrt{ \frac{\frac{\sigma^2_{u} + \sigma^2_{wg}}{n} + \sigma^2}{rt} + \frac{\sigma^2_{ge}}{t} + \sigma^2_{g} }} \\
\end{split}
\end{equation}

From Equation \ref{eq:genetic_gain}, it is clear that improvements in genetic gain per year can
be made by increasing the number of replications and/or environments tested, increasing the 
number of plants within each experimental plot, or increasing overall selection pressure 
during the breeding program. Genomic selection improves genetic gain by addressing 
all of these at once.

\begin{itemize}
    \item A breeder may grow many more plants than can be evaluated using field plots in 
          early generations. This allows a program to increase the overall selection 
          differential without reducing the number of new lines at the end of the breeding pipeline.
    \item Because genomic prediction estimate genetic potential directly from genetic information,
          genotype by environment interaction is reduced to zero.
    \item In most scenarios, only one SNP assay is run per sample. When data collection is properly
          calibrated and quality controlled this results in near zero measurement error. This reduces
          within-plot and between-plot (or more accurately, within genotype and between genotype) 
          variation from all sources to zero.
\end{itemize}

However, unlike the genetic gain equation, genomic prediction introduces prediction error to the 
denominator of the equation. By comparing the magnitude of prediction error with the cost
of executing early-generation SNP assays in a breeding program, it is possible to determine
if genomic selection is beneficial to a breeding program. This thesis explores the application 
of regularized neural networks to minimizing genomic selection prediction error. 






